disabled-tests-inspector/src/main/java/io/quarkus/ts/DisabledTestAnalyserService.java

```diff
@@ -0,0 +1,217 @@
+                }
+
+                if (reason == null) {
+                    reason = getDisablingReasonComment(previousLine);
+                }
+
+                if (reason == null) {
+                    reason = getDisablingReasonComment(currentLine);
+                }
+
+                annotationTypes.add(annotationType);
+                reasons.add(reason);
+                issueLinks.add(issueLink);
+            }
+
+            if (insideDisabledBlock && (testMethodMatcher.lookingAt() || (classMatcher.lookingAt()))) {
+                for (int i = 0; i < annotationTypes.size(); i++) {
+                    disabledTests.add(new DisabledTest(
+                            currentTestMethod != null ? currentTestMethod : "All methods",
+                            currentClass,
+                            annotationTypes.get(i),
+                            reasons.get(i),
+                            issueLinks.get(i),
+                            fileUrl,
+                            isGitHubIssueClosed(issueLinks.get(i))
+                    ));
+
+                    String moduleName = extractModuleName(ghContent.getPath());
+                    moduleStats.putIfAbsent(moduleName, new DisabledTestsModuleStats());
+                    moduleStats.get(moduleName).incrementAnnotation(annotationTypes.get(i));
+                }
+
+                annotationTypes.clear();
+                reasons.clear();
+                issueLinks.clear();
+                insideDisabledBlock = false;
+            }
+            previousLine = currentLine;
+        }
+        return disabledTests;
+    }
+
+    private String extractIssueLink(String text) {
+        Matcher issueMatcher = ISSUE_PATTERN.matcher(text);
+        return issueMatcher.find() ? issueMatcher.group(1) : null;
+    }
+
+    private String getDisablingReasonComment(String line) {
+        if (line.contains("//")) {
+            return line.substring(line.indexOf("//") + 2).trim();
+        }
+        return null;
+    }
+
+    private boolean isGitHubIssueClosed(String issueLink) {
+        if (issueLink != null && issueLink.contains("github.com")) {
+            try {
+                String[] parts = issueLink.split("/");
+                String repo = parts[parts.length - 4] + "/" + parts[parts.length - 3];
+                int issueNumber = Integer.parseInt(parts[parts.length - 1]);
+
+                GitHub github = GitHub.connect();
+                GHIssue issue = github.getRepository(repo).getIssue(issueNumber);
+                return issue.getState() == GHIssueState.CLOSED;
+            } catch (Exception e) {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    private String extractModuleName(String filePath) {
+        return filePath.substring(0, filePath.indexOf("/src"));
+    }
+
+    private String getStatsOrTestFileName(String fileName, String branchName, boolean isStats) {
```

Comment:
I also do this, but passing boolean into method to change behavior is not a good pattern. (I even read it somewhere, no idea where). IMO this would be better pattern:

```
private String getTestFileName(String fileName, String branchName) {
        return getFileName(fileName, branchName, "");
    }

private String getStatsFileName(String fileName, String branchName) {
    return getFileName(fileName, branchName, "-stats");
}
private String getFileName(String fileName, String branchName, String fileNameSuffix) {
    return fileName.replaceFirst("(\\.json)?$", "-" + branchName + fileNameSuffix + ".json");
}
```